eslint/babel-eslint-parser/src/configuration.js

```diff
@@ -30,19 +30,35 @@ export function normalizeBabelParseConfig(options) {
     exclude: options.babelOptions.exclude,
     ignore: options.babelOptions.ignore,
     only: options.babelOptions.only,
-    parserOpts: {
-      allowImportExportEverywhere: options.allowImportExportEverywhere,
-      allowReturnOutsideFunction: true,
-      allowSuperOutsideMethod: true,
-      ranges: true,
-      tokens: true,
-      plugins: ["estree"],
-    },
+    parserOpts: normalizeBabelParserOptions(options),
     caller: {
       name: "@babel/eslint-parser",
     },
   };
 
+  function normalizeBabelParserOptions(options) {
+    const parserOpts = options.babelOptions.parserOpts || {};
+    const parserPlugins = parserOpts.plugins || [];
+
+    const normalizedParserOptions = Object.assign(
+      {
+        allowReturnOutsideFunction: true,
+        allowSuperOutsideMethod: true,
+        ranges: true,
+        tokens: true,
+      },
+      parserOpts,
+    );
+
+    normalizedParserOptions.plugins = ["estree", ...parserPlugins];
+    if (options.allowImportExportEverywhere !== undefined) {
```

Comment:
So if `options.allowImportExportEverywhere` exists, it'll use that over `parserOpts. allowImportExportEverywhere` to prevent breaking changes.